cpp/examples/minimal_abm.cpp

```diff
@@ -0,0 +1,158 @@
+    auto shop = world.add_location(mio::abm::LocationType::BasicsShop);
+    world.get_individualized_location(shop).get_infection_parameters().set<mio::abm::MaximumContacts>(20);
+    // At every school, the maximum contacts are 20.
+    auto school = world.add_location(mio::abm::LocationType::School);
+    world.get_individualized_location(school).get_infection_parameters().set<mio::abm::MaximumContacts>(20);
+     // At every workplace, maximum contacts are 10.
+    auto work = world.add_location(mio::abm::LocationType::Work);
+    world.get_individualized_location(work).get_infection_parameters().set<mio::abm::MaximumContacts>(10);
+
+    // People can get tested at work (and do this with 0.5 probability).
+    auto testing_min_time = mio::abm::days(1);
+    auto probability      = 0.5;
+    auto start_date       = mio::abm::TimePoint(0);
+    auto end_date         = mio::abm::TimePoint(0) + mio::abm::days(60);
+    auto test_type        = mio::abm::AntigenTest();
+    auto test_at_work = std::vector<mio::abm::LocationType>{mio::abm::LocationType::Work};
+    auto testing_criteria_work =
+        std::vector<mio::abm::TestingCriteria>{mio::abm::TestingCriteria({}, test_at_work, {})};
+    auto testing_scheme_work =
+        mio::abm::TestingScheme(testing_criteria_work, testing_min_time, start_date, end_date, test_type, probability);
+    world.get_testing_strategy().add_testing_scheme(testing_scheme_work);
+
+    // Assign locations to the people
+    for (auto& person : persons) {
+        //assign shop and event
+        person.set_assigned_location(event);
+        person.set_assigned_location(shop);
+        //assign hospital and ICU
+        person.set_assigned_location(hospital);
+        person.set_assigned_location(icu);
+        //assign work/school to people depending on their age
+        if (person.get_age() == mio::abm::AgeGroup::Age5to14) {
+            person.set_assigned_location(school);
+        }
+        if (person.get_age() == mio::abm::AgeGroup::Age15to34 || person.get_age() == mio::abm::AgeGroup::Age35to59) {
+            person.set_assigned_location(work);
+        }
+    }
+
+    // During the lockdown, social events are closed for 90% of people. 
+    auto t_lockdown = mio::abm::TimePoint(0) + mio::abm::days(10);
+    mio::abm::close_social_events(t_lockdown, 0.9, world.get_migration_parameters());
+
+    auto t0         = mio::abm::TimePoint(0);
+    auto tmax       = mio::abm::TimePoint(0) + mio::abm::days(30);
+    auto sim = mio::abm::Simulation(t0, std::move(world));
+    sim.advance(tmax);
+
+    // The results are saved in a table with 9 rows.
+    // The first row is t = time, the others correspond to the number of people with a certain infection state at this time:
+    // S = Susceptible, E = Exposed, C= Carrier, I= Infected, I_s = Infected_Severe,
+    // I_c = Infected_Critical, R_C = Recovered_Carrier, R_I = Recovered_Infected, D = Dead
```

Comment:
Please adapt to new infection state names.

Reply:
Is this related to issue #387 ?

Reply:
Yes but I was wrong. It is not yet merged, David's still working on. So maybe we change this later.